test/typescript/modules.test.ts

```diff
@@ -72,6 +72,7 @@ i18next.use(externalModules);
 
 // exercise class usage
 class MyLoggerModule implements LoggerModule {
+  static type: 'logger' = 'logger';
   type: 'logger' = 'logger';
```

Comment:
Unfortunate there is a duplication. However, they mean different things. The `static` one ensures the constructor has a property called `type`. And the original one ensures the instances have the said property. I fixed the overall issue in the easy way. The hard way is to acknowledge that `type` is only needed for that `.use()` method invocation and should separate this field from the `Module` interface for good. However, the problem is how can we make sure someone who makes a `LanguageDetector` will end up assigning the `type` field to be `languageDetector` when we separate them. That's the hard part and I don't have an easy solution for it.

Reply:
Nonetheless, if this change is approved. All other repos need also to duplicate the `type` field in their TypeScript definition. It seems it doesn't hurt break functionalities though, since all of them are implemented in JavaScript. But it will break those who use TypeScript.

Reply:
This is the 100% correct interpretation of the existing runtime.  It seems rather inconvenient but there are tradeoffs made for flexibility and usage.  Given the frequency of usage is several magnitudes the number of module coded, it makes sense to keep the runtime signatures as-is and use this duplicative typescript use in subordinate modules.

With that said, add a comment to this particular file that simply states:
> Need both static and member definitions of `type` to satisfy `use()` signature, see #1442